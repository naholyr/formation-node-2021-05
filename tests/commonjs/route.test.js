const { routeFibo } = require("./route.js");

const mockFibo = jest.fn(); // spy

jest.mock("./fibo.js", () => {
  return {
    fibo: (n) => mockFibo(n),
  };
});

describe("Fibo (route)", () => {
  it("should call fibo", () => {
    mockFibo.mockImplementation(() => 42);

    const req = {
      params: {
        number: "10",
      },
    };

    const res = {
      send: jest.fn(),
    };

    routeFibo(req, res);

    expect(mockFibo).toHaveBeenCalledWith(10);
    expect(res.send).toHaveBeenCalledWith({ input: 10, output: 42 });
  });

  it("should match snapshot", () => {
    const foo = () => {
      return { toto: 43 };
    };
    expect(foo()).toMatchSnapshot();
  });
});
