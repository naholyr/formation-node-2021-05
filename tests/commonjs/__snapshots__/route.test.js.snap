// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fibo (route) should match snapshot 1`] = `
Object {
  "toto": 43,
}
`;
